Add gtag consent mode v0 support

tomalec opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->


Add the minimalistic implementation to add support for [Google Consent mode](https://developers.google.com/tag-platform/security/guides/consent?consentmode=advanced)

We want to add MVP ASAP before Google starts penalizing merchants without this setup. (pcTzPl-25o-p2)

This PR adds a basic default and documents how to customize/extend it.


### Screenshots:

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17435/81dadd01-38fe-4565-aff1-a6aed5614fde)


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Build, install, activate the extension
2. Deactivate Google Analytics Integration extension
3. Open [Tag assistant](https://tagassistant.google.com/)
4. Visit shop page
5. You should see consent mode set up
   ![image](https://github.com/woocommerce/google-listings-and-ads/assets/17435/7342e3dc-53ab-48be-846a-44d8fd58ba6c)
7. Read the docs, check if they are explanatory enough
8. Try the following snippet
```php
add_filter(
    'woocommerce_gla_gtag_consent',
    function( $old_config ) {
        $additional_config = "
        gtag( 'consent', 'default', {
            analytics_storage: 'granted',
            ad_storage: 'granted',
            ad_user_data: 'denied',
            ad_personalization: 'denied',
            region: ['PL'], // <--------- SET IT TO YOUR REGION
        } );
        ";
        return $old_config . $additional_config;
    }
);
```
9. You should see another consent entry  specified
   ![image](https://github.com/woocommerce/google-listings-and-ads/assets/17435/3c629c52-cd25-4f4b-9a26-b4be49ad7a17)




### Additional details:

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Add - Google Analytics consent mode support.

tomalec committed:
Add gtag consent mode v0 support

martynmjones commented on docs/gtag-consent-mode.md:
```diff
@@ -0,0 +1,7 @@
+
+To respect your customers' privacy, we set up the default state of [consent mode](https://support.google.com/analytics/answer/9976101). We set it to deny all the parameters for visitors from the EEA region. You can add an extension or CMP that delivers a banner or any other UI to let visitors update their consent in runtime.
+
+You can also customize your own default state configuration using the `woocommerce_gla_gtag_consent` filter providing any snippet that uses [Google's `gtag('consent', 'default', {…})` API ](https://developers.google.com/tag-platform/security/guides/consent?consentmode=advanced).
```
It could be useful to mention that the consent state can be updated like you did in the [Google Analytics extension docs](https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/d8c750cb182013489e7de44db547a61f3f3b61fd/README.md#consent-mode).

martynmjones commented on docs/gtag-consent-mode.md:
```diff
@@ -0,0 +1,7 @@
+## Googla Analytics (gtag) Consent Mode
+
+Unless you're running the [Google Analytics Integration](https://woo.com/products/woocommerce-google-analytics/) extension for a more sophisticated configuration, Google Listings and Ads will add Google's `gtag` to help you track some customer behavior.
```
Should we be using `Google Analytics for WooCommerce` here to reflect the updated branding?

martynmjones reviewed (approved):
Hey @tomalec, thanks for your work on consent in GLA!

I've left two comments on the docs but all is working well so LGTM ✅

tomalec committed:
Mention how to update the consent mode in runtime

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/2290#discussion_r1510985763

tomalec committed:
Update WCGAI to GA4W branding

Address https://github.com/woocommerce/google-listings-and-ads/pull/2290/files#r1510986691
